File: src/transform.js
function run(input) {
  const events = (input && input.data && input.data.events) || [];
  const now = new Date();
  const today = toLocalDateString(now);
  const tomorrow = toLocalDateString(
    new Date(now.getFullYear(), now.getMonth(), now.getDate() + 1)
  );

  const today_events = [];
  const tomorrow_events = [];

  for (const event of events) {
    const dt = new Date(event.date_time);
    if (isNaN(dt.getTime())) continue;
    const localDate = toLocalDateString(dt);

    const mapped = {
      title: event.summary || "",
      time: event.all_day ? "All day" : (event.start || ""),
      all_day: !!event.all_day,
    };

    if (localDate === today) {
      today_events.push(mapped);
    } else if (localDate === tomorrow) {
      tomorrow_events.push(mapped);
    }
  }

  return { today_events, tomorrow_events };
}

function toLocalDateString(d) {
  const y = d.getFullYear();
  const m = String(d.getMonth() + 1).padStart(2, "0");
  const day = String(d.getDate()).padStart(2, "0");
  return `${y}-${m}-${day}`;
}

File: src/transform.py
from datetime import datetime, date, timedelta


def run(input):
    events = input.get("data", {}).get("events", [])
    today = date.today()
    tomorrow = today + timedelta(days=1)

    today_events = []
    tomorrow_events = []

    for event in events:
        dt_str = event.get("date_time", "")
        try:
            dt = datetime.fromisoformat(dt_str.replace("Z", "+00:00"))
            local_date = dt.astimezone().date()
        except (ValueError, AttributeError):
            continue

        mapped = {
            "title": event.get("summary", ""),
            "time": "All day" if event.get("all_day") else event.get("start", ""),
            "all_day": event.get("all_day", False),
        }

        if local_date == today:
            today_events.append(mapped)
        elif local_date == tomorrow:
            tomorrow_events.append(mapped)

    return {
        "today_events": today_events,
        "tomorrow_events": tomorrow_events,
    }
